### PR TITLE
[Native] Allow '$file_size' and '$file_modified_time' for HiveSplits in SQL queries

### PR DESCRIPTION
**Pull request by aditi-pandit**
Fixes https://github.com/prestodb/presto/issues/21867

## Test Plan
e2e tests

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by aditi-pandit** [comment]
@Yuhta : This is the Prestissimo side change for info columns. PTAL.

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 1]
```diff
@@ -948,12 +948,23 @@ std::shared_ptr<connector::ConnectorTableHandle> toConnectorTableHandle(
...
+    }
+
+    // Add synthesized columns to the TableScanNode columnHandles as well.
+    for (const auto& entry : hiveLayout->predicateColumns) {
+      if (toHiveColumnType(entry.second.columnType) ==
+          velox::connector::hive::HiveColumnHandle::ColumnType::kSynthesized) {
+        if (assignments.count(entry.first) == 0) {
+          assignments.emplace(
+              entry.first, toColumnHandle(&entry.second, typeParser));
+        }
```
Comment (Yuhta) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:966]:
```suggestion
    for (const auto& [name, col] : hiveLayout->predicateColumns) {
      if (toHiveColumnType(col.columnType) ==
          velox::connector::hive::HiveColumnHandle::ColumnType::kSynthesized) {
        VELOX_CHECK(assignments.emplace(name, toColumnHandle(&col, typeParser)).second, "Duplicate assignment: {}", name);
```

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 2]
```diff
@@ -983,10 +994,21 @@ std::shared_ptr<connector::ConnectorTableHandle> toConnectorTableHandle(
           std::dynamic_pointer_cast<const protocol::IcebergTableLayoutHandle>(
               tableHandle.connectorTableLayout)) {
     for (const auto& entry : icebergLayout->partitionColumns) {
-      partitionColumns.emplace(
+      assignments.emplace(
```
Comment (Yuhta) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:997]: Abstract these to a function and reuse

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp (hunk 1 above)
Comment (aditi-pandit) [reply presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:966]: @Yuhta : We want to add these to the assignments list only if they are not present already. Its not quite the same code.

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp (hunk 2 above)
Comment (aditi-pandit) [reply presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:1002]: Done.

**Comment by aditi-pandit** [comment]
@Yuhta : PTAL.

File: presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp [hunk 3]
```diff
@@ -948,12 +948,28 @@ std::shared_ptr<connector::ConnectorTableHandle> toConnectorTableHandle(
     const VeloxExprConverter& exprConverter,
     const TypeParser& typeParser,
     std::unordered_map<std::string, std::shared_ptr<connector::ColumnHandle>>&
-        partitionColumns) {
+        assignments) {
+  auto addSynthesizedColumn = [&](const std::string& name,
+                                  protocol::ColumnType columnType,
+                                  const protocol::ColumnHandle& column) {
+    if (toHiveColumnType(columnType) ==
+        velox::connector::hive::HiveColumnHandle::ColumnType::kSynthesized) {
```
Comment (majetideepak) [inline presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp:956]: nit: `connector::hive::HiveColumnHandle::ColumnType` should be sufficient.

**Review by majetideepak — APPROVED** [review]

**Pull request merged**